Patterns/Validation.md

```diff
@@ -45,3 +48,43 @@ public class Data
 ```
 
 The `[ValidateEnumerable]` and `[ValidateObject]` attributes instruct the validator to go into the enumerable or object. Error messages from the deeper properties are grouped under the error message of the `Children` and `Parent` properties. The implementation of these attributes can be found in NETDefault/Libs/Triple.DataAnnotations.
+
+### Validating dates
+
+Validating dates in a concise way can be achieved using specific JsonConverter classes. In the above example you can find the property `StartDate` of type `DateTime`. The `JsonConverter` makes use of the class `DateConverter`:
+
+```c#
+public class DateConverter : IsoDateTimeConverter
```

Comment:
Maybe interesting to extend the sample with this.

```
[Required, JsonConverter(typeof(DateConverter), "MM-dd;yyyy")]
public DateTime StartDate { get; set; }
```

```
public class DateConverter : IsoDateTimeConverter
    {
        public DateConverter(string format)
        {
            DateTimeFormat = format;
        }

        public DateConverter()
        {
            DateTimeFormat = "yyyy-MM-dd";
        }

        public override object? ReadJson(JsonReader reader, Type objectType, object? existingValue, JsonSerializer serializer)
        {
            try
            {
                return base.ReadJson(reader, objectType, existingValue, serializer);
            }
            catch
            {
                return null;
            }
        }
    }
```

Reply:
The only issue with this approach is that the validation is a bit crude. If a client send "2022-01-01" to a DateConverter with format "yyyy-MM" the base IsoDateTimeConverter fails with an exception (that's why the `base.ReadJson` is in a try-catch). But this exception causes the DateConverter to return `null`, which will default to DateTime.MinValue for non-nullable DateTime properties. 

So it's good to make note of this and advice to add a Range attribute as well when validating DateTimes, so the DateTime.MinValue is caught.

Reply:
I tried to find a way to combine a jsonConverter with an ValidationAtrribute but since JsonConverterAttribute is sealed this was not something I could easily do.. maybe its possible to add a custom attribute that adds the two attribute types to the property.. but so far I had no success.

Reply:
It's going to be hard as you need to validate before deserialization -- something that is only possible by deserializing the request model to some other class (with only primitive properties) first. That kind of overhead is probably too much when you can solve it by providing decent API documentation.